examples/contact/beam_bending.cpp

```diff
@@ -0,0 +1,112 @@
+
+#include "serac/physics/solid_mechanics_contact.hpp"
+#include "serac/infrastructure/terminator.hpp"
+#include "serac/mesh/mesh_utils.hpp"
+#include "serac/physics/state/state_manager.hpp"
+#include "serac/physics/materials/parameterized_solid_material.hpp"
+#include "serac/serac_config.hpp"
+
+int main(int argc, char* argv[])
+{
+  serac::initialize(argc, argv);
+
+  // NOTE: p must be equal to 1
+  constexpr int p = 1;
+  // NOTE: dim must be equal to 3
+  constexpr int dim = 3;
+
+  // Create DataStore
+  std::string            name = "contact_beam_example";
+  axom::sidre::DataStore datastore;
+  serac::StateManager::initialize(datastore, name + "_data");
+
+  // Construct the appropriate dimension mesh and give it to the data store
+  std::string filename = SERAC_REPO_DIR "/data/meshes/beam-hex-with-contact-block.mesh";
+
+  auto mesh = serac::mesh::refineAndDistribute(serac::buildMeshFromFile(filename), 2, 0);
+  serac::StateManager::setMesh(std::move(mesh), "beam_mesh");
+
+  serac::LinearSolverOptions linear_options{.linear_solver = serac::LinearSolver::Strumpack, .print_level = 1};
+#ifndef MFEM_USE_STRUMPACK
+  SLIC_INFO_ROOT("Contact requires MFEM built with strumpack.");
+  return 1;
+#endif
+
+  serac::NonlinearSolverOptions nonlinear_options{.nonlin_solver  = serac::NonlinearSolver::Newton,
+                                                  .relative_tol   = 1.0e-12,
+                                                  .absolute_tol   = 1.0e-12,
+                                                  .max_iterations = 200,
+                                                  .print_level    = 1};
+
+  serac::ContactOptions contact_options{.method      = serac::ContactMethod::SingleMortar,
+                                        .enforcement = serac::ContactEnforcement::Penalty,
+                                        .type        = serac::ContactType::Frictionless,
+                                        .penalty     = 1.0e3};
+
+  serac::SolidMechanicsContact<p, dim, serac::Parameters<serac::L2<0>, serac::L2<0>>> solid_solver(
+      nonlinear_options, linear_options, serac::solid_mechanics::default_quasistatic_options,
+      serac::GeometricNonlinearities::On, name, "beam_mesh", {"bulk_mod", "shear_mod"});
+
+  serac::FiniteElementState K_field(serac::StateManager::newState(serac::L2<0>{}, "bulk_mod", "beam_mesh"));
+  mfem::Vector              K_values({10.0, 10.0, 100.0});
+  mfem::PWConstCoefficient  K_coeff(K_values);
+  K_field.project(K_coeff);
+  solid_solver.setParameter(0, K_field);
+
+  serac::FiniteElementState G_field(serac::StateManager::newState(serac::L2<0>{}, "shear_mod", "beam_mesh"));
+  mfem::Vector              G_values({0.25, 0.25, 2.5});
+  mfem::PWConstCoefficient  G_coeff(G_values);
+  G_field.project(G_coeff);
+  solid_solver.setParameter(1, G_field);
+
+  serac::solid_mechanics::ParameterizedNeoHookeanSolid<dim> mat{1.0, 0.0, 0.0};
+  solid_solver.setMaterial(serac::DependsOn<0, 1>{}, mat);
```

Comment:
can we add some description of these different coefficient values? Are there 3 bodies in contact or just 2 (and if so, which values correspond to which bodies)?

This is good motivation for us to adopt the `Domain` at the physics module level